Sources/ArgumentParser/Parsing/Interactive.swift

```diff
@@ -57,51 +57,91 @@ extension CommandParser {
+        try update(InputOrigin(elements: [.interactive]), name, string, &values)
+      }
       
-      let input: [String]
+      // All possible strings that can be converted to value
+      // of this CaseIterable enum type.
       let allValues = definition.help.allValues
       if allValues.isEmpty {
-        // Get normal value
-        input = getNormalValue(label: label)
+        storeNormalValues(label: label, updateBy: updateBy, arguments: arguments)
       } else {
         allValues.enumerated().forEach { print("\($0 + 1). \($1)") }
-        // Get CaseIterable Enum
-        input = getCaseIterableEnum(label: label, allValues: allValues)
-      }
-      
-      // Split array input like "1 2 3".
-      for element in input {
-        try update(InputOrigin(elements: [.interactive]), name, element, &values)
+        storeCaseIterableEnums(label: label, allValues: allValues, updateBy: updateBy, arguments: arguments)
       }
-      
+
       return true
         
     default: break
     }
     
     return false
   }
-  
-  fileprivate mutating func getNormalValue(label: String) -> [String] {
+
+  fileprivate mutating func storeNormalValues(label: String,
+                                              updateBy: (String) throws -> Void,
+                                              arguments: ArgumentSet)
+  {
     print("? Please enter '\(label)': ", terminator: "")
-    return getInput()?.components(separatedBy: " ") ?? [""]
+    let strs = getInput()?.components(separatedBy: " ") ?? [""]
+
+    for str in strs {
+      do {
+        try updateBy(str)
+      } catch {
+        // Handle ParserError
+        guard let error = error as? ParserError else { return }
+        let generator = ErrorMessageGenerator(arguments: arguments, error: error)
+        let description = generator.makeErrorMessage() ?? error.localizedDescription
+        guard case let .unableToParseValue(_, _, original, _, _) = error else { return }
+        print("Error: " + description + ".\n")
+        replaceInvalidValue(original: original, updateBy: updateBy, arguments: arguments)
+      }
+    }
   }
-  
-  fileprivate mutating func getCaseIterableEnum(label: String, allValues: [String]) -> [String] {
+
+  fileprivate mutating func replaceInvalidValue(original: String,
+                                                updateBy: (String) throws -> Void,
+                                                arguments: ArgumentSet)
+  {
+    print("? Please replace '\(original)': ", terminator: "")
+    let input = getInput() ?? ""
+
+    do {
+      try updateBy(input)
+    } catch {
+      // Handle ParserError
+      guard let error = error as? ParserError else { return }
+      let generator = ErrorMessageGenerator(arguments: arguments, error: error)
+      let description = generator.makeErrorMessage() ?? error.localizedDescription
+      guard case let .unableToParseValue(_, _, original, _, _) = error else { return }
```

Comment:
The code will be clearer if we group the two preconditions together. Since they have the same failure outcome (an early `return`), you can combine them into a single `guard` statement, as well.